### PR TITLE
Fix action for Tag and Release

### PR DESCRIPTION
**Pull request by atchox**
### New
- use `CPack` for packaging. Packaging for `*-Darwin-arm64.tar.gz` (usable via `brew`) and `*-Linux-x86_64.deb` for now.

### Fixes
- update brew taps before installing
- pull version info from tag during build

### CODE REVIEW

File: .vscode/settings.json
Comment (imallona) [inline .vscode/settings.json:1]: This is not part of the codebase, perhaps added by mistake? I'd remove it

File: .vscode/sftp.json
Comment (imallona) [inline .vscode/sftp.json:1]: This is not part of the codebase, perhaps added by mistake? I'd remove it

Comment (atchox) [reply .vscode/sftp.json:1]: removed

File: .vscode/settings.json
Comment (atchox) [reply .vscode/settings.json:1]: removed

**Comment by imallona** [comment]
looks awesome, great job!

I confirm the deb works fine on a x86_64 debian-like.

There are some unresolved comments above, could you please close them when fixed or reasoned why not updating them? Main concerns are checks on gcc versions; and releasing (deb, brew tarballing) only after tagging/releasing, not on every push

File: method/CMakeLists.txt [hunk 1]
```diff
@@ -2,7 +2,12 @@
 cmake_minimum_required(VERSION 3.15)
 
 # Project name and version
-project(yamet VERSION 0.1.0 LANGUAGES CXX)
+project(yamet VERSION 0.0.0 LANGUAGES CXX)
```
[the remark this answers is not in the document]
Comment (atchox) [reply method/CMakeLists.txt:5]: This is a default value for local development. It uses the version specified in the tag when you push a new tag.

**Pull request merged**